lib/stdio/pipe.js

```diff
@@ -0,0 +1,18 @@
+// When the `std*: Iterable | WebStream | URL | filePath`, `input` or `inputFile` option is used, we pipe to `childProcess.std*`.
+// When the `std*: Array` option is used, we emulate some of the native values ('inherit', Node.js stream and file descriptor integer). To do so, we also need to pipe to `childProcess.std*`.
+// Therefore the `std*` options must be either `pipe` or `overlapped`. Other values do not set `childProcess.std*`.
+export const updateStdio = stdioStreamsGroups => stdioStreamsGroups.map(stdioStreams => updateStdioItem(stdioStreams));
+
+// Whether `childProcess.std*` will be set
+export const willPipeStreams = stdioStreams => PIPED_STDIO_VALUES.has(updateStdioItem(stdioStreams));
+
+const PIPED_STDIO_VALUES = new Set(['pipe', 'overlapped', undefined, null]);
+
+const updateStdioItem = stdioStreams => {
+	if (stdioStreams.length > 1) {
+		return stdioStreams.some(({value}) => value === 'overlapped') ? 'overlapped' : 'pipe';
+	}
+
+	const [stdioStream] = stdioStreams;
+	return stdioStream.type !== 'native' && stdioStream.value !== 'overlapped' ? 'pipe' : stdioStream.value;
```

Comment:
When using transforms, we set the `std*` option passed to `child_process.spawn()` as `pipe`. This allows piping transforms.

The `encoding: 'hex'` and `encoding: 'base64'` options use an internal transform. The bug is: this internal transform currently sets the `std*` option to `pipe`, due to the above, even when something like `stdout: 'ignore'` has been used.

For example, `{ stdout: 'ignore', encoding: 'base64' }` would become `{ stdout: 'pipe', encoding: 'base64' }`.

However, there is no need to add the `encoding` internal transform when the stream is ignored. This is the solution implemented by this PR.